plonky2x/src/frontend/builder/mod.rs

```diff
@@ -168,11 +169,35 @@ impl<L: PlonkParameters<D>, const D: usize> CircuitBuilder<L, D> {
         V::init(self)
     }
 
+    /// Initializes an array of variables with no value in the circuit.
+    pub fn init_array<V: CircuitVariable, const N: usize>(&mut self) -> ArrayVariable<V, N> {
+        ArrayVariable::init(self)
+    }
+
+    /// Initializes an array of variables with no value in the circuit without any validity checks.
+    pub fn init_array_unsafe<V: CircuitVariable, const N: usize>(&mut self) -> ArrayVariable<V, N> {
+        ArrayVariable::init_unsafe(self)
+    }
+
     /// Initializes a variable with a constant value in the circuit.
     pub fn constant<V: CircuitVariable>(&mut self, value: V::ValueType<L::Field>) -> V {
         V::constant(self, value)
     }
 
+    /// Initializes an array of variables with a constant value in the circuit.
+    pub fn constant_array<V: CircuitVariable, const N: usize>(
+        &mut self,
+        value: &[V::ValueType<L::Field>; N],
```

Comment:
nit: why don't you just put in a slice here instead of requiring a fixed length slice?